### PR TITLE
Added a pre-push hook that will run flow, lint, and test scripts

### PR DESCRIPTION
**Pull request by ackinc**
Relevant issue: NA
Tested in browser: NA

Runs the flow, lint, and unit test commands before every `git push`
Mainly useful for WIP PRs - simple errors will now be caught locally instead of in the CI stage

### CODE REVIEW

**Comment by andytuba** [comment]
Haven't read the docs, but I assume husky will add a prepush git hook when it's installed by npm/yarn?

**Comment by ackinc** [comment]
Yes; and husky can add other hooks too - pre-commit etc. - depending on what we put in package.json

**Comment by ackinc** [comment]
I don't think the hook is added when husky is installed, since we're able to change the hook (in package.json or the husky config file) long after installation.

I'll try to dig into the library a bit to see how it works

**Comment by ackinc** [comment]
Looked into it - husky does add git hooks (all hooks) at installation time

File: package.json [hunk 1]
```diff
@@ -108,6 +113,7 @@
     "firefox-extension-deploy": "1.1.1",
     "flow-bin": "0.82.0",
     "html-loader": "0.5.5",
+    "husky": "^1.1.4",
```
Comment (mc10) [inline package.json]: This should be a fixed version.

Comment (ackinc) [reply package.json]: I'll make that change. Does it also make sense to add a line to CONTRIBUTING.md, at the top-level or as a sub-section under the 'Adding New Files' section, saying something like "When adding a new dependency or devDependency, make sure to fix the version in package.json"?

Comment (andytuba) [reply package.json]: We should make a linter that runs in CI and pre push :grin:

Comment (andytuba) [reply package.json]: And also document it to contributing.md

**Pull request merged**